### PR TITLE
Improved linkchecking

### LINKED ISSUE #319: [FEATURE] Improved resource (and general) linkchecking
**Describe your suggested improvement**
Currently, linkchecking for resources is done by a bespoke script, which works well and fails runs that have broken links.

Looking into linkchecking in general, I think that there is room for use of tools like [Lychee](https://github.com/lycheeverse/lychee-action) to run linkchecking on the whole codebase (in fact, the example action they have also automagically creates issues in the codebase for broken links, which would be super neat for surfacing these issues).

**Describe the benefit**
Working links and resources = happy users

**Would you be capable/willing to implement the improvement?**
Likely yes

**Additional comments**
Currenlty in the brainstorming/exploration phase. Please discuss before submitting a PR.

### PR DESCRIPTION
**Pull request by VeckoTheGecko**
## Describe your changes

Added [Lychee](https://github.com/lycheeverse/lychee-action) linkchecking action workflow to the repo to go through repo files looking for broken links (404s etc). The workflow:

- Runs Lychee, checking for broken links. Generates a report with findings. Workflow always passes regardless of broken links
- If broken links found:
  - Automatically updates #322 with the body of the report, and re-opens the issue if it was closed
- If no broken links found:
  - Automatically updates #322 with a corresponding statement and closes the issue

Issue #322 in this case acts as the main mechanism for surfacing these broken links, in a way that is minimally intrusive (doesn't mess with GHA failures) and interested parties can subscribe to the issue.

## Related issue number/link

Fixes #319

### CODE REVIEW

**Comment by dmlb** [comment]
This is cool :D

File: .github/workflows/url-checker.yml [hunk 1]
```diff
@@ -1,22 +1,58 @@
...
     - cron: 0 0 * * 0
+env:
+  LINK_TRACKING_ISSUE_NUMBER: 322
+  LYCHEE_REPORT_FILE: ./lychee/out.md
+permissions:
+  issues: write
 
 jobs:
-  check_urls:
+  linkChecker:
```
Comment (dmlb) [inline .github/workflows/url-checker.yml:14]: does the mixed casing matter on this repo?
file contains snake_case, camelCase and kebab-case

**Review by dmlb — APPROVED** [review]

File: .github/workflows/url-checker.yml (hunk 1 above)
Comment (VeckoTheGecko) [reply .github/workflows/url-checker.yml:14]: Not at all. A remnant of ChatGPT that I was a bit too lazy to fix 😁. Fixed now

**Comment by VeckoTheGecko** [comment]
> This is cool :D

I thought so too 😁 . Really nice way to manage the links.

Thanks for the review @dmlb !

@all-contributors please add @dmlb for review

**Comment by VeckoTheGecko** [comment]
> I've put up [a pull request](https://github.com/ClimateTown/knowledge-hub/pull/326) to add `@thanks`! 🎉

What the heck

**Pull request merged**